Dockerfile build fails

I got an error on `docker build . -t myImageName`
I'm on Mac M1 arch, I don't know if it's related

```
 => ERROR [build 7/7] RUN make -j4                                                                                                                                                                                                                20.7s 
------                                                                                                                                                                                                                                                  
 > [build 7/7] RUN make -j4:                                                                                                                                                                                                                            
0.257 perl golpe/gen-fbs.pl                                                                                                                                                                                                                             
0.258 perl golpe/gen-config.pl                                                                                                                                                                                                                          
0.258 golpe/external/rasgueadb/rasgueadb-generate golpe.yaml build                                                                                                                                                                                      
0.261 perl golpe/gen-main.cpp.pl                                                                                                                                                                                                                        
0.284 perl golpe/gen-golpe.h.pl
0.360 perl golpe/external/templar/templar.pl src/tmpls/ strfrytmpl build/StrfryTemplates.h
0.371 cd golpe/external/uWebSockets && make -j libuWS.a
0.381 make[1]: warning: -j0 forced in submake: resetting jobserver mode.
0.382 make[1]: Entering directory '/build/golpe/external/uWebSockets'
0.382 g++ -std=c++17 -O2 -g -Wall -fPIC    -c -o src/Extensions.o src/Extensions.cpp
0.383 g++ -std=c++17 -O2 -g -Wall -fPIC    -c -o src/Group.o src/Group.cpp
0.383 g++ -std=c++20 -O3 -g -Wall -fPIC  -DDOCOPT_HEADER_ONLY -Iinclude -Ibuild -Isrc -Igolpe/external -Igolpe/external/lmdbxx/include -Igolpe/external/config/include -Igolpe/external/json/include -Igolpe/external/PEGTL/include -Igolpe/external/hoytech-cpp -Igolpe/external/docopt.cpp -Igolpe/external/loguru -Igolpe/external/parallel-hashmap -Iexternal/negentropy/cpp -MMD -MP -MT golpe/logging.o -MF golpe/logging.d -c golpe/logging.cpp -o golpe/logging.o
0.385 g++ -std=c++17 -O2 -g -Wall -fPIC    -c -o src/Networking.o src/Networking.cpp
0.386 g++ -std=c++20 -O3 -g -Wall -fPIC  -DDOCOPT_HEADER_ONLY -Iinclude -Ibuild -Isrc -Igolpe/external -Igolpe/external/lmdbxx/include -Igolpe/external/config/include -Igolpe/external/json/include -Igolpe/external/PEGTL/include -Igolpe/external/hoytech-cpp -Igolpe/external/docopt.cpp -Igolpe/external/loguru -Igolpe/external/parallel-hashmap -Iexternal/negentropy/cpp -MMD -MP -MT build/main.o -MF build/main.d -c build/main.cpp -o build/main.o
0.386 g++ -std=c++17 -O2 -g -Wall -fPIC    -c -o src/Hub.o src/Hub.cpp
0.390 g++ -std=c++17 -O2 -g -Wall -fPIC    -c -o src/Node.o src/Node.cpp
0.397 g++ -std=c++17 -O2 -g -Wall -fPIC    -c -o src/WebSocket.o src/WebSocket.cpp
0.412 g++ -std=c++17 -O2 -g -Wall -fPIC    -c -o src/HTTPSocket.o src/HTTPSocket.cpp
0.419 g++ -std=c++17 -O2 -g -Wall -fPIC    -c -o src/Socket.o src/Socket.cpp
0.428 g++ -std=c++17 -O2 -g -Wall -fPIC    -c -o src/Epoll.o src/Epoll.cpp
0.438 g++ -std=c++17 -O2 -g -Wall -fPIC    -c -o src/Room.o src/Room.cpp
0.709 g++ -std=c++20 -O0 -g -Wall -fPIC  -DDOCOPT_HEADER_ONLY -Iinclude -Ibuild -Isrc -Igolpe/external -Igolpe/external/lmdbxx/include -Igolpe/external/config/include -Igolpe/external/json/include -Igolpe/external/PEGTL/include -Igolpe/external/hoytech-cpp -Igolpe/external/docopt.cpp -Igolpe/external/loguru -Igolpe/external/parallel-hashmap -Iexternal/negentropy/cpp -MMD -MP -MT build/config.o -MF build/config.d -c build/config.cpp -o build/config.o
5.414 ar rs libuWS.a src/Extensions.o src/Group.o src/Networking.o src/Hub.o src/Node.o src/WebSocket.o src/HTTPSocket.o src/Socket.o src/Epoll.o src/Room.o
5.422 ar: creating libuWS.a
5.452 make[1]: Leaving directory '/build/golpe/external/uWebSockets'
5.452 g++ -std=c++20 -O3 -g -Wall -fPIC  -DDOCOPT_HEADER_ONLY -Iinclude -Ibuild -Isrc -Igolpe/external -Igolpe/external/lmdbxx/include -Igolpe/external/config/include -Igolpe/external/json/include -Igolpe/external/PEGTL/include -Igolpe/external/hoytech-cpp -Igolpe/external/docopt.cpp -Igolpe/external/loguru -Igolpe/external/parallel-hashmap -Iexternal/negentropy/cpp -MMD -MP -MT src/Decompressor.o -MF src/Decompressor.d -c src/Decompressor.cpp -o src/Decompressor.o
12.21 g++: fatal error: Killed signal terminated program cc1plus
12.21 compilation terminated.
12.21 make: *** [golpe/rules.mk:31: build/config.o] Error 1
12.21 make: *** Waiting for unfinished jobs....
------
Dockerfile:13
--------------------
  11 |     RUN git submodule update --init
  12 |     RUN make setup-golpe
  13 | >>> RUN make -j4
  14 |     
  15 |     FROM ubuntu:jammy as runner
--------------------
ERROR: failed to solve: process "/bin/sh -c make -j4" did not complete successfully: exit code: 2
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,9 +1,14 @@
 # Built by Akito
 # npub1wprtv89px7z2ut04vvquscpmyfuzvcxttwy2csvla5lvwyj807qqz5aqle
 
-FROM alpine:3.18.3 as build
+FROM alpine:3.18.3 AS build
+
 ENV TZ=Europe/London
+
 WORKDIR /build
+
+COPY . .
+
 RUN \
   apk --no-cache add \
     linux-headers \
@@ -13,7 +18,6 @@ RUN \
     pkgconfig \
     libtool \
     ca-certificates \
-    perl-yaml-libyaml \
     perl-yaml \
     perl-template-toolkit \
     perl-app-cpanminus \
@@ -24,14 +28,13 @@ RUN \
     libsecp256k1-dev \
     zstd-dev \
   && rm -rf /var/cache/apk/* \
-  && cpanm Regexp::Grammars
-
-COPY . .
-RUN git submodule update --init
-RUN make setup-golpe
-RUN make -j4
+  && cpanm Regexp::Grammars \
+  && git submodule update --init \
+  && make setup-golpe \
+  && make -j4
 
 FROM alpine:3.18.3
+
 WORKDIR /app
 
 RUN \
@@ -41,9 +44,12 @@ RUN \
     libsecp256k1 \
     libb2 \
     zstd \
-    libressl3.7-libssl \
+    libressl \
   && rm -rf /var/cache/apk/*
 
 COPY --from=build /build/strfry strfry
+
+EXPOSE 7777
+
 ENTRYPOINT ["/app/strfry"]
 CMD ["relay"]

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,23 +1,48 @@
-FROM ubuntu:jammy as build
+# Built by Akito
+# npub1wprtv89px7z2ut04vvquscpmyfuzvcxttwy2csvla5lvwyj807qqz5aqle
+
+FROM alpine:3.18.3 as build
 ENV TZ=Europe/London
 WORKDIR /build
-RUN apt update && apt install -y --no-install-recommends \
-    git g++ make pkg-config libtool ca-certificates \
-    libyaml-perl libtemplate-perl libregexp-grammars-perl libssl-dev zlib1g-dev \
-    liblmdb-dev libflatbuffers-dev libsecp256k1-dev \
-    libzstd-dev
+RUN \
+  apk --no-cache add \
+    linux-headers \
+    git \
+    g++ \
+    make \
+    pkgconfig \
+    libtool \
+    ca-certificates \
+    perl-yaml-libyaml \
+    perl-yaml \
+    perl-template-toolkit \
+    perl-app-cpanminus \
+    libressl-dev \
+    zlib-dev \
+    lmdb-dev \
+    flatbuffers-dev \
+    libsecp256k1-dev \
+    zstd-dev \
+  && rm -rf /var/cache/apk/* \
+  && cpanm Regexp::Grammars
 
 COPY . .
 RUN git submodule update --init
 RUN make setup-golpe
 RUN make -j4
 
-FROM ubuntu:jammy as runner
+FROM alpine:3.18.3
 WORKDIR /app
 
-RUN apt update && apt install -y --no-install-recommends \
-    liblmdb0 libflatbuffers1 libsecp256k1-0 libb2-1 libzstd1 \
-    && rm -rf /var/lib/apt/lists/*
+RUN \
+  apk --no-cache add \
+    lmdb \
+    flatbuffers \
+    libsecp256k1 \
+    libb2 \
+    zstd \
+    libressl3.7-libssl \
+  && rm -rf /var/cache/apk/*
 
 COPY --from=build /build/strfry strfry
 ENTRYPOINT ["/app/strfry"]

--- a/ubuntu.Dockerfile
+++ b/ubuntu.Dockerfile
@@ -1,0 +1,24 @@
+FROM ubuntu:jammy as build
+ENV TZ=Europe/London
+WORKDIR /build
+RUN apt update && apt install -y --no-install-recommends \
+    git g++ make pkg-config libtool ca-certificates \
+    libyaml-perl libtemplate-perl libregexp-grammars-perl libssl-dev zlib1g-dev \
+    liblmdb-dev libflatbuffers-dev libsecp256k1-dev \
+    libzstd-dev
+
+COPY . .
+RUN git submodule update --init
+RUN make setup-golpe
+RUN make -j4
+
+FROM ubuntu:jammy as runner
+WORKDIR /app
+
+RUN apt update && apt install -y --no-install-recommends \
+    liblmdb0 libflatbuffers1 libsecp256k1-0 libb2-1 libzstd1 \
+    && rm -rf /var/lib/apt/lists/*
+
+COPY --from=build /build/strfry strfry
+ENTRYPOINT ["/app/strfry"]
+CMD ["relay"]

